packages/lodestar/src/chain/clock/LocalClock.ts

```diff
@@ -0,0 +1,69 @@
+import {Epoch, Slot} from "@chainsafe/lodestar-types";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {computeEpochAtSlot, getCurrentSlot} from "@chainsafe/lodestar-beacon-state-transition";
+
+import {ChainEventEmitter} from "../emitter";
+
+import {IBeaconClock} from "./interface";
+
+/**
+ * A local clock, the clock time is assumed to be trusted
+ */
+export class LocalClock implements IBeaconClock {
+  private readonly config: IBeaconConfig;
+  private readonly genesisTime: number;
+  private timeoutId: NodeJS.Timeout;
+  private readonly signal: AbortSignal;
+  private readonly emitter: ChainEventEmitter;
+  private _currentSlot: number;
+
+  public constructor({
+    config,
+    genesisTime,
+    emitter,
+    signal,
+  }: {
+    config: IBeaconConfig;
+    genesisTime: number;
+    emitter: ChainEventEmitter;
+    signal: AbortSignal;
+  }) {
+    this.config = config;
+    this.genesisTime = genesisTime;
+    this.timeoutId = setTimeout(this.onNextSlot, this.msUntilNextSlot());
+    this.signal = signal;
+    this.emitter = emitter;
+    this._currentSlot = getCurrentSlot(this.config, this.genesisTime);
+    signal.addEventListener("abort", this.abort);
```

Comment:
https://github.com/mysticatea/event-target-shim/blob/master/src/event-target.mjs#L202

There is this param option, where you can set `once: true` to avoid removing event listener.
cc @dapplion since he asked me this ^^

Reply:
Nice find! Thanks

Reply:
Keep in mind that if signal don't get triggered, we still need to remove manually